python/usb_console/radio_session.py

```diff
@@ -59,27 +72,78 @@ def check_for_downlink(self):
         Read device output for debug messages and state variable updates. Record debug messages
         to the logging file, and update the console's record of the state.
         '''
+        
+        #connect to PAN email account  
+        mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+        mail.login(self.username, self.password)
+        mail.select('Inbox')
 
         while self.running_logger:
-            # TODO implement
-            pass
+            #.search() searches from mail. Data gives id's of all emails.
+            type, data = mail.search(None, '(FROM "sbdservice@sbd.iridium.com")', '(UNSEEN)')
+            mail_ids = data[0]
+            id_list = mail_ids.split()
+
+            for num in id_list:
+                #.fetch() fetches the mail for given id where 'RFC822' is an Internet 
+                # Message Access Protocol.
+                typ, data = mail.fetch(num,'(RFC822)')
+                
+                #go through each component of data and print sender and subject line
+                for response_part in data:
+                    if isinstance(response_part, tuple):
+                        # converts message from byte literal to string removing b''
+                        msg = email.message_from_string(response_part[1].decode('utf-8'))
+                        #get the file attached to the message
+                        for part in msg.walk():
+                            
+                            if part.get_content_maintype() == 'multipart':
+                                continue
+
+                            if part.get('Content-Disposition') is None:
+                                continue
+                            
+                            #check if there is an attachment
+                            if part.get_filename() is not None:
+                                #get json data from email attachment
+                                attachmentContents=part.get_payload(decode=True).decode('utf8')
+                                data=json.loads(attachmentContents)
+                                
+                                #iterate through each statefield key of the dictionary
+                                logline=""
```

Comment:
Could you instead log the state field value in the datastore? The logger is a good place to put debugging data (e.g. "Connected to Email!", "Disconnected from email.") but the datastore should be the destination for any telemetry collected by a Session.

When you put a value into the datastore, the value should be dict with two keys: "time", which is the datetime of the received value, and "val", which is the actual value.